### ISSUE TITLE
Determine Drug/Drug Family Functionality in CTS API v2

### ISSUE THREAD

**Issue by sarinapadilla** [NCIOCPL/clinical-trials-search-app#523]
Initially, there was an issue with the current API request used to populate the autosuggest on the Drug/Drug Family field on the Advanced form. This functionality was changed from v1 to v2. CTRP has confirmed that they have implemented the fix to allow our app to use the same API request as previously.

This ticket is to confirm that the existing requests works with API v2 as expected. If not, then we need to let CTRP know that the issue has not been resolved.

**Comment by alerosad** [comment]
Making https://clinicaltrialsapi.cancer.gov/api/v2/interventions?category=Agent&category=Agent%20Category&name=ant as a request through Postman to the API v2 returns 200 success, when passed in an api key.

Taking the request params from the URL and passing them to the Intervention section of https://clinicaltrialsapi.cancer.gov/doc (category=Agent, name=ant) it returns a 200 along with the data. The request URL goes as follows: https://clinicaltrialsapi.cancer.gov/api/v2/interventions?name=ant&category=agent

With this said, it seems like the V2 version works with the given request URL when using Postman or the swagger docs.

**Comment by alerosad** [comment]
After taking https://clinicaltrialsapi.cancer.gov/v1/diseases?code[]=C8286&current_trial_status[]=Active&current_trial_status[]=Approved&current_trial_status[]=Enrolling+by+Invitation&current_trial_status[]=In+Review&current_trial_status[]=Temporarily+Closed+to+Accrual&current_trial_status[]=Temporarily+Closed+to+Accrual+and+Intervention&size=100&type[]=maintype&type[]=subtype&type[]=stage&type[]=finding 

and sending the request to v2, as: 

http://localhost:3000/cts/proxy-api/v2/diseases?code[]=C8286&current_trial_status[]=Active&current_trial_status[]=Approved&current_trial_status[]=Enrolling+by+Invitation&current_trial_status[]=In+Review&current_trial_status[]=Temporarily+Closed+to+Accrual&current_trial_status[]=Temporarily+Closed+to+Accrual+and+Intervention&size=100&type[]=maintype&type[]=subtype&type[]=stage&type[]=finding

Both requests return similar looking payloads, with the only difference being in V2. In V2 there is the "synonyms"  param, which isn't found in V1. Here is the data from both requests:

V1: {
    "terms": [
        {
            "name": "AIDS-Related Burkitt Lymphoma",
            "codes": [
                "C8286"
            ],
            "ancestor_ids": [
                "C27134",
                "C3208",
                "C3211"
            ],
            "parent_ids": [
                "C3211"
            ],
            "type": [
                "subtype"
            ]
        }
    ]
}

V2: {
    "data": [
        {
            "name": "AIDS-Related Burkitt Lymphoma",
            "codes": [
                "C8286"
            ],
            "ancestor_ids": [
                "C27134",
                "C3208",
                "C3211"
            ],
            "type": [
                "subtype"
            ],
            "parent_ids": [
                "C3211"
            ],
            "synonyms": [
                "AIDS Related Burkitt's Lymphoma",
                "AIDS Related Small Non Cleaved Cell Lymphoma",
                "AIDS-Associated Burkitt's Lymphoma",
                "AIDS-Associated Small Non Cleaved Cell Lymphoma",
                "AIDS-Related Burkitt's Lymphoma",
                "AIDS-Related Small Non-Cleaved Cell Lymphoma"
            ],
            "count": 1
        }
    ]
}

In conclusion, it seems that V1 requests do work on V2 without having to adjust the request params

**Comment by alerosad** [comment]
With an advanced search with the Params of Drug and Treament being "Ibrutinib" and "Piezoelectric Breast Palpatation". Sending the V1 request with these drug params, to a v2 request yields:
```
V1: {
    "terms": [
        {
            "name": "Ibrutinib",
            "codes": [
                "C81934"
            ],
            "synonyms": [
                "Imbruvica"
            ],
            "category": "agent",
            "count": 52
        },
        {
            "name": "Piezoelectric Breast Palpation",
            "codes": [
                "C177735"
            ],
            "synonyms": [
                "iBreastExam",
                "iBreastExam Scan",
                "iBreastExam Test"
            ],
            "category": "other",
            "count": 1
        }
    ]
}
```
```
V2:{
    "data": [
        {
            "name": "Ibrutinib",
            "codes": [
                "C81934"
            ],
            "category": [
                "agent"
            ],
            "type": [
                "drug"
            ],
            "synonyms": [],
            "count": 52
        },
        {
            "name": "Piezoelectric Breast Palpation",
            "codes": [
                "C177735"
            ],
            "category": [
                "other"
            ],
            "type": [
                "procedure/surgery"
            ],
            "synonyms": [],
            "count": 1
        }
    ]
}
```


V1 request URL: https://clinicaltrialsapi.cancer.gov/v1/interventions?code[]=C81934&code[]=C177735&current_trial_status[]=Active&current_trial_status[]=Approved&current_trial_status[]=Enrolling+by+Invitation&current_trial_status[]=In+Review&current_trial_status[]=Temporarily+Closed+to+Accrual&current_trial_status[]=Temporarily+Closed+to+Accrual+and+Intervention&size=100&sort=name&order=asc


V2 request URL: http://localhost:3000/cts/proxy-api/v2/interventions?code[]=C81934&code[]=C177735&current_trial_status[]=Active&current_trial_status[]=Approved&current_trial_status[]=Enrolling+by+Invitation&current_trial_status[]=In+Review&current_trial_status[]=Temporarily+Closed+to+Accrual&current_trial_status[]=Temporarily+Closed+to+Accrual+and+Intervention&size=100&sort=name&order=asc


With this said, inputting Drug and Treatment values in the advanced search yields a similar response from V1 and V2, while having the same request being sent for both versions of the API

**Comment by sarinapadilla** [comment]
Closing this ticket, as the functionality that we need for Drug/Drug Family has been confirmed.
